### DISCUSSION TITLE
Repository has unknown error: Another open merge request already exists for this source branch

### DISCUSSION THREAD

**Discussion by Jonnymcc** [renovatebot/renovate#17397] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.174.1

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab Enterprise Edition 14.0.11-ee

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Renovate-bot is running in Kubernetes and the pod always exits in error. Not sure what the error is since it appears to process all repos. There is an error with "Response code 409 (Conflict)" which could be causing the pod to exit with error code 1.

```
    State:          Terminated
      Reason:       Error
      Exit Code:    1
```

The error message includes "Another open merge request already exists for this source branch", but I'm not sure if this is expected to cause renovate-bot to exit in error. Seems like something to warn about, but could be ignored.

### Relevant debug logs

<details><summary>Logs</summary>

```
{
           "name": "renovate",
           "level": 50,
           "logContext": "5-83V3IBsl-NGpSwmmaXV",
           "repository": "REDACTED",
           "err": {
             "name": "HTTPError",
             "code": "ERR_NON_2XX_3XX_RESPONSE",
             "timings": {
               "start": 1661369639892,
               "socket": 1661369639893,
               "lookup": 1661369639905,
               "connect": 1661369639974,
               "secureConnect": 1661369640108,
               "upload": 1661369640108,
               "response": 1661369640307,
               "end": 1661369640307,
               "phases": {
                 "wait": 1,
                 "dns": 12,
                 "tcp": 69,
                 "tls": 134,
                 "request": 0,
                 "firstByte": 199,
                 "download": 0,
                 "total": 415
               }
             },
             "message": "Response code 409 (Conflict)",
             "stack": "HTTPError: Response code 409 (Conflict)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
             "options": {
               "headers": {
                 "user-agent": "RenovateBot/32.174.1 (https://github.com/renovatebot/renovate)",
                 "private-token": "**redacted**",
                 "accept": "application/json",
                 "content-type": "application/json",
                 "content-length": "2799",
                 "accept-encoding": "gzip, deflate, br"
               },
               "url": "REDACTED",
               "hostType": "gitlab",
               "username": "",
               "password": "",
               "method": "POST",
               "http2": false
             },
             "response": {
               "statusCode": 409,
               "statusMessage": "Conflict",
               "body": {
                 "message": [
                   "Another open merge request already exists for this source branch: !7"
                 ]
               },
               "headers": {
                 "server": "nginx",
                 "date": "Wed, 24 Aug 2022 19:34:00 GMT",
                 "content-type": "application/json",
                 "content-length": "84",
                 "connection": "close",
                 "cache-control": "no-cache",
                 "vary": "Origin",
                 "x-content-type-options": "nosniff",
                 "x-frame-options": "SAMEORIGIN",
                 "x-request-id": "REDACTED",
                 "x-runtime": "0.130065"
               },
               "httpVersion": "1.1"
             }
           },
           "msg": "Repository has unknown error"
         }
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
It means that a different account has created an MR from Renovate's branch. Was the bot renamed/moved?

Note that this doesn't cause the bot to exit immediately, if just causes it to use an error code when it eventually exits

**Reply by viceice** [reply]
we've recently another discussion about a flag to not let renovate exit with non-zero status. But can't find it.

**Comment by viceice** [comment, marked as answer]
I think renovate works as expected here

**Reply by rarkins** [reply]
Yes, converted to a discussion.

If Renovate cannot create PRs because a user it doesn't recognize as itself has created a PR from Renovate's branch, then an error is logged.

When errors are logged, Renovate exits with a non-zero exit code after processing all repos.

**Comment by Jonnymcc** [comment]
How do I resolve the error or get renovate-bot to exit with a zero exit code?

**Reply by rarkins** [reply]
Find the repo and find the MR which has been opened for the branch in question

**Reply by Jonnymcc** [reply]
Do I have to close the MR or delete the branch, or both?

**Reply by rarkins** [reply]
Close and delete to be safe, but it would be useful for you to first try to work out why it happened. For example if it's because someone else is running their own Renovate Bot in the same repo, you'll have problems again soon after if they don't stop.
